File: clash/rotate-180/solution.hs
trans :: Char -> Char
trans c
  | c == 'b' = 'q' | c == 'q' = 'b'
  | c == 'u' = 'n' | c == 'n' = 'u'
  | c == 'p' = 'd' | c == 'd' = 'p'
  | otherwise = c

solve :: String -> String
solve = reverse . map trans

main :: IO ()
main = interact $ solve

File: clash/rotate-180/solution.clj
(ns Solution
  (:gen-class))

(defn trans [c]
  (case c
    \b \q
    \q \b
    \u \n
    \n \u
    \p \d
    \d \p
    c))

(defn -main [& args]
  (let [input (read-line)
        ans (apply str (reverse (map trans input)))]
    (println ans)))